subgraphs/collectible-market/mappings/index.ts

```diff
@@ -0,0 +1,263 @@
+  transaction.collection = event.params.collection.toHex();
+  transaction.tokenId = event.params.collection.toString() + "-" + event.params.tokenId.toString();
+  transaction.askPrice = event.params.price.toBigDecimal().div(EIGHTEEN_BD);
+  transaction.netPrice = event.params.netPrice.toBigDecimal().div(EIGHTEEN_BD);
+  transaction.buyer = event.params.buyer.toHex();
+  transaction.seller = event.params.seller.toHex();
+  transaction.withBNB = event.params.withBNB;
+
+  // 2. Buyer
+  let buyer = User.load(event.params.buyer.toHex());
+
+  // Buyer may not exist
+  if (buyer == null) {
+    buyer = new User(event.params.buyer.toHex());
+    buyer.numberTokensListed = ZERO_BI;
+    buyer.numberTokensPurchased = ONE_BI;
+    buyer.numberTokensSold = ZERO_BI;
+    buyer.totalVolumeInBNBTokensPurchased = event.params.price.toBigDecimal().div(EIGHTEEN_BD);
+    buyer.totalVolumeInBNBTokensSold = ZERO_BD;
+    buyer.totalFeesCollectedInBNB = ZERO_BD;
+    buyer.averageTokenPriceInBNBPurchased = event.params.price.toBigDecimal().div(EIGHTEEN_BD);
+    buyer.averageTokenPriceInBNBSold = ZERO_BD;
+  } else {
+    buyer.numberTokensPurchased = buyer.numberTokensPurchased.plus(ONE_BI);
+    buyer.totalVolumeInBNBTokensPurchased = buyer.totalVolumeInBNBTokensPurchased.plus(
+      event.params.price.toBigDecimal().div(EIGHTEEN_BD)
+    );
+    buyer.averageTokenPriceInBNBPurchased = buyer.totalVolumeInBNBTokensPurchased.div(
+      buyer.numberTokensPurchased.toBigDecimal()
+    );
+  }
+
+  // 3. Seller
+  let seller = User.load(event.params.seller.toHex());
+
+  seller.numberTokensSold = seller.numberTokensSold.plus(ONE_BI);
+  seller.totalVolumeInBNBTokensSold = seller.totalVolumeInBNBTokensSold.plus(
+    event.params.netPrice.toBigDecimal().div(EIGHTEEN_BD)
+  );
+  seller.averageTokenPriceInBNBSold = seller.totalVolumeInBNBTokensSold.div(seller.numberTokensSold.toBigDecimal());
+
+  // 4. NFT
+  let tokenConcatId = event.params.collection.toString() + "-" + event.params.tokenId.toString();
+  let token = NFT.load(tokenConcatId);
+
+  token.latestTradedPriceInBNB = event.params.price.toBigDecimal().div(EIGHTEEN_BD); // divDecimal
+  token.tradeVolumeBNB = token.tradeVolumeBNB.plus(event.params.price.toBigDecimal().div(EIGHTEEN_BD));
+  token.totalTrades = token.totalTrades.plus(ONE_BI);
+
+  token.isTradable = false;
+
+  transaction.save();
+  buyer.save();
+  seller.save();
+  token.save();
+}
+
+/**
+ * REVENUE CLAIMS BY CREATOR/TREASURY
+ */
+
+export function handleRevenueClaim(event: RevenueClaim): void {
+  let user = User.load(event.params.claimer.toHex());
+
+  if (user == null) {
+    user = new User(event.params.claimer.toHex());
+    user.numberTokensListed = ZERO_BI;
+    user.numberTokensPurchased = ZERO_BI;
+    user.numberTokensSold = ZERO_BI;
+    user.totalVolumeInBNBTokensPurchased = ZERO_BD;
+    user.totalVolumeInBNBTokensSold = ZERO_BD;
+    user.totalFeesCollectedInBNB = event.params.amount.toBigDecimal().div(EIGHTEEN_BD);
```

Comment:
```suggestion
    user.totalFeesCollectedInBNB = toBigDecimal(event.params.amount, 18);
```
Same as for other previous comments.